cortex-m-pac/src/lib.rs

```diff
@@ -0,0 +1,264 @@
+#![no_std]
+
+pub mod result;
+
+use result::Result;
+
+/// Trait for enums of target-specific exception numbers.
+///
+/// This trait should be implemented by a peripheral access crate (PAC) on its enum of available
+/// exceptions for a specific device. Alternatively, the `cortex-m` crate provides a default
+/// implementation. Each variant must convert to a `usize` of its exception number.
+///
+/// # Safety
+///
+/// * This trait must only be implemented on the `cortex-m` crate or on a PAC of a Cortex-M target.
+/// * This trait must only be implemented on enums of exceptions.
+/// * Each enum variant must represent a distinct value (no duplicates are permitted),
+/// * Each enum variant must always return the same value (do not change at runtime).
+/// * All the exception numbers must be less than or equal to `MAX_EXCEPTION_NUMBER`.
+/// * `MAX_EXCEPTION_NUMBER` must coincide with the highest allowed exception number.
+pub unsafe trait ExceptionNumber: Copy {
+    /// Highest number assigned to an exception.
+    const MAX_EXCEPTION_NUMBER: usize;
```

Comment:
It seems strange to define MAX_EXCEPTION_NUMBER on each instance of ExceptionNumber.